ipc/cli/src/commands/node/init.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: MIT
+
+use crate::comet_runner::run_comet;
+use crate::commands::node::config::{GenesisSource, NodeInitConfig};
+use crate::commands::node::config_override::merge_toml_config;
+use crate::commands::subnet::join::join_subnet;
+use crate::{
+    get_ipc_provider, ipc_config_store::IpcConfigStore, CommandLineHandler, GlobalArguments,
+};
+use anyhow::{Context, Ok};
+use async_trait::async_trait;
+use clap::Args;
+use fendermint_app::cmd::config::write_default_settings as write_default_fendermint_setting;
+use fendermint_app::cmd::genesis::into_tendermint;
+use fendermint_app::options::genesis::GenesisIntoTendermintArgs;
+
+use fendermint_app::cmd::key::store_key;
+use fendermint_crypto::SecretKey;
+use fs_err as fs;
+use ipc_api::subnet_id::SubnetID;
+use ipc_provider::IpcProvider;
+use std::path::{Path, PathBuf};
+
+use crate::commands::subnet::create_genesis::create_genesis;
+use crate::commands::wallet::import::{import_wallet, WalletImportArgs};
+
+pub(crate) struct InitNode;
+
+#[async_trait]
+impl CommandLineHandler for InitNode {
+    type Arguments = InitNodeArgs;
+
+    async fn handle(global: &GlobalArguments, arguments: &Self::Arguments) -> anyhow::Result<()> {
+        let ipc_config_store = IpcConfigStore::load_or_init(global).await?;
+        let config = NodeInitConfig::load(&arguments.config)?;
+
+        let subnet_id: SubnetID = config.subnet.parse().context("invalid subnet ID")?;
+        let parent_id: SubnetID = config.parent.parse().context("invalid parent ID")?;
+
+        let home = Path::new(&config.home);
+        create_dir(home)?;
+        let fendermint_home = home.join("fendermint");
+        create_dir(&fendermint_home)?;
+        let comet_bft_home = home.join("cometbft");
+        create_dir(&comet_bft_home)?;
+
+        let provider = get_ipc_provider(global)?;
+        log::info!("Importing and storing validator key");
+        import_and_store_validator_key(&provider, &config.key, &fendermint_home)?;
+        log::info!("Validator key imported and stored");
+
+        if let Some(join_config) = &config.join {
+            let mut provider = get_ipc_provider(global)?;
+            log::info!("Joining subnet `{}` as `{}`", subnet_id, join_config.from);
+            let args = join_config.clone().into_args(subnet_id.to_string());
+            join_subnet(&mut provider, &args).await?;
+            log::info!("Joined subnet `{}` as `{}`", subnet_id, join_config.from);
+        };
+
+        let created_genesis = match config.genesis {
+            GenesisSource::Create(gen_cfg) => {
+                let parent = ipc_config_store
+                    .get_subnet(&parent_id)
+                    .await
+                    .context("parent subnet not found in config store")?;
+
+                let created_genesis =
+                    create_genesis(&gen_cfg, &subnet_id, &parent, &fendermint_home).await?;
```

Comment:
I'm currently failing here with the following log entry
```
2025-07-29T20:34:19.110068Z  INFO ipc_cli::commands::subnet::create_genesis: Creating genesis    
2025-07-29T20:34:21.927259Z ERROR ipc_cli: main process failed: error processing command Some(Node(NodeCommandsArgs { command: Init(InitNodeArgs { config: "./conf.yaml" }) })): Invalid name: please ensure the contract and method you're calling exist! failed to decode empty bytes. if you're using jsonrpc this is likely due to jsonrpc returning `0x` in case contract or method don't exist    
```

the `~/.ipc/config.toml` contains the default values from `ipc-cli config init`.

Reply:
It seems like you're trying to run int without deploying contracts or the subnet existing?

Reply:
The `~/.ipc/config.toml` default values suppose to target an already deployed gateway/registry, no?